tests/bud.bats

```diff
@@ -662,6 +662,41 @@ _EOF
   expect_output --substring "Cache burst add diff"
 }
 
+@test "bud --layers with --mount type bind should preserve cache when file mod time changes but content stays same" {
+  _prefetch alpine
+  local contextdir=${TEST_SCRATCH_DIR}/bud/platform
+  mkdir -p $contextdir
+
+  cat > $contextdir/samplefile << _EOF
+samplefile content unchanged
+_EOF
+
+  cat > $contextdir/Containerfile << _EOF
+FROM alpine
+RUN --mount=type=bind,source=samplefile,target=file,Z cat file
+_EOF
+
+  # on first run since there is no cache so content must be printed
+  run_buildah build $WITH_POLICY_JSON --layers -t source -f $contextdir/Containerfile $contextdir
+  expect_output --substring "samplefile content unchanged"
+
+  # on second run since there is cache so content should not be printed
+  run_buildah build $WITH_POLICY_JSON --layers -t source -f $contextdir/Containerfile $contextdir
+  # output should not contain content from the file since entire build is cached
+  assert "$output" !~ "samplefile content unchanged"
+
+  # Remove and recreate the file with same content (this changes modification time)
+  rm $contextdir/samplefile
+  cat > $contextdir/samplefile << _EOF
+samplefile content unchanged
+_EOF
```

Comment:
Using `touch $contextdir/samplefile` would be simpler here, though if it wasn't being zeroed out, it wouldn't guarantee that the timestamp had changed enough to affect the digest of the tarstream that would be generated, so the test might end up passing anyway.  Perhaps `touch -d @...` after creating the file, and then again with a different date to "update" it?

Reply:
I updated the PR.